src/tests/mechs.rs

```diff
@@ -25,6 +25,22 @@ fn test_get_mechs() {
     );
     assert_eq!(ret, CKR_OK);
     assert_eq!(true, count > 4);
+    #[cfg(feature = "no_sha1")]
+    {
+        let mut sha1_found = 0;
+        for mech in &mechs {
+            sha1_found += match &*mech {
+                &CKM_SHA1_RSA_PKCS => 1,
+                &CKM_ECDSA_SHA1 => 1,
+                &CKM_SHA1_RSA_PKCS_PSS => 1,
+                &CKM_SHA_1_HMAC => 1,
+                &CKM_SHA_1_HMAC_GENERAL => 1,
+                &CKM_SHA_1 => 1,
```

Comment:
`CKM_SHA_1_KEY_GEN` missing here from what I grepped in the code.

Reply:
Note that we do not implement all of these, but you can check for them regardless, they should never be there if no_sha1 is selected anyway